core/domain/state_domain.py

```diff
@@ -0,0 +1,543 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain object for state."""
```

Comment:
Ah, we should put all the constituents of state here as well. I.e. Interaction, AnswerGroups, etc. -- basically, everything that a state consists of.

(Please modify the fileoverview docstring accordingly.)

Reply:
Done.